src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,717 @@
+    code_ += doubleSpacing + "\treturn {{VALUENAME}}(fbb, o: tableOffset)";
+    code_ += doubleSpacing + "}";
+    code_ += spacing + "}";
+    code_ += spacing + "return nil";
+    code_ += "\t}";
+  }
+
+  std::string GenOffset() { return "let o = {{ACCESS}}.offset({{OFFSET}}); "; }
+
+  std::string GenReaderMainBody(std::string optional = "") {
+    return "\tpublic var {{VALUENAME}}: {{VALUETYPE}}" + optional + " { ";
+  }
+  std::string GenReader(std::string type, std::string at = "{{OFFSET}}") {
+    return "{{ACCESS}}.readBuffer(of: {{" + type + "}}.self, at: " + at + ")";
+  }
+
+  std::string GenArrayMainBody(std::string optional = "") {
+    return "\tpublic func {{VALUENAME}}(at index: Int32) -> {{VALUETYPE}}" +
+           optional + " { ";
+  }
+
+  std::string GenIndirect(std::string reading) {
+    return "{{ACCESS}}.indirect(" + reading + ")";
+  }
+
+  std::string GenConstructor(
+      std::string offset = "{{ACCESS}}.postion + {{OFFSET}}") {
+    return "{{VALUETYPE}}({{ACCESS}}.bb, o: " + offset + ") }";
+  }
+
+  std::string GenMutate(std::string offset, std::string getOffset = "",
+                        bool isRaw = false) {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}) -> Bool {" +
+           getOffset + " return {{ACCESS}}.mutate({{VALUENAME}}" +
+           (isRaw ? ".rawValue" : "") + ", index: " + offset + ") }";
+  }
+
+  std::string GenMutateArray() {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}, at index: "
+           "Int32) -> Bool { " +
+           GenOffset() +
+           "return {{ACCESS}}.directMutate({{VALUENAME}}, index: "
+           "{{ACCESS}}.vector(at: o) + index * {{SIZE}}) }";
+  }
+
+  std::string GenEnumBaseContructor() {
+    return "{{VALUETYPE}}(rawValue: {{CONSTANT}})!";
+  }
+
+  std::string GenEnumConstructor(std::string at = "{{OFFSET}}") {
+    return "{{VALUETYPE}}(rawValue: " + GenReader("BASEVALUE", at) + ") ";
+  }
+
+  std::string GenEndEnumConstructor() {
+    return "?? {{VALUETYPE}}(rawValue: {{DEFAULT}})! }";
+  }
+
+  std::string GenType(const Type &type) const {
+    return IsScalar(type.base_type)
+               ? GenTypeBasic(type)
+               : (IsArray(type) ? GenType(type.VectorType())
+                                : GenTypePointer(type));
+  }
+
+  std::string GenTypePointer(const Type &type) const {
+    switch (type.base_type) {
+      case BASE_TYPE_STRING:
+        return "String";
+      case BASE_TYPE_VECTOR:
+        return GenType(type.VectorType());
+      case BASE_TYPE_STRUCT:
+        return WrapInNameSpace(*type.struct_def);
+      case BASE_TYPE_UNION:
+      default:
+        return "FlatBufferObject";
+    }
+  }
+
+  std::string GenTypeBasic(const Type &type) const {
+    return GenTypeBasic(type, true);
+  }
+
+  std::string GenTypeBasic(const Type &type, bool CanOverride) const {
+    // clang-format off
+        static const char * const swift_type[] = {
+#define FLATBUFFERS_TD(ENUM, IDLTYPE, \
```

Comment:
indent pre-processor statements. Surround with `// clang-format: off` and `on` if this was caused by `clang-format`.